### PR TITLE
Fix for LNURLp with ZBD wallet 

### PR DESCRIPTION
**Pull request by bitkarrot**
Possible fix to https://github.com/lnbits/lnbits/issues/2457

modify description_hash/unhashed_desc for lnurlp payments

### CODE REVIEW

File: lnbits/wallets/zbd.py [hunk 1]
```diff
@@ -64,18 +65,25 @@ async def create_invoice(
...
         # https://api.zebedee.io/v0/charges
-        if description_hash or unhashed_description:
-            raise UnsupportedError("description_hash")
-
+        
         msats_amount = amount * 1000
         data: Dict = {
             "amount": f"{msats_amount}",
-            "description": memo,
+            "description": "",
```
Comment (dni) [inline lnbits/wallets/zbd.py:72]:
```suggestion
```

File: lnbits/wallets/zbd.py [hunk 2]
```diff
@@ -64,18 +65,25 @@ async def create_invoice(
...
         }
 
+        ## handle description_hash and unhashed for ZBD
+        if description_hash:
+            data["description"] = description_hash.hex()
+        elif unhashed_description:
+            data["description"] = hashlib.sha256(unhashed_description).hexdigest()
+        else: 
+            desc = memo or ""
+            data["description"] = desc
```
Comment (dni) [inline lnbits/wallets/zbd.py:85]:
```suggestion
            data["description"] = memo or ""
```

File: lnbits/wallets/zbd.py [hunk 3]
```diff
@@ -64,18 +65,25 @@ async def create_invoice(
...
             "expiresIn": 3600,
             "callbackUrl": "",
             "internalId": "",
         }
 
+        ## handle description_hash and unhashed for ZBD
+        if description_hash:
+            data["description"] = description_hash.hex()
+        elif unhashed_description:
+            data["description"] = hashlib.sha256(unhashed_description).hexdigest()
```
Comment (prusnak) [inline lnbits/wallets/zbd.py:82]: other sources use `descriptionHash` for the hashes - I think we should do the same here?

```suggestion
            data["descriptionHash"] = description_hash.hex()
        elif unhashed_description:
            data["descriptionHash"] = hashlib.sha256(unhashed_description).hexdigest()
```

**Review by prusnak — CHANGES_REQUESTED** [review]

File: lnbits/wallets/zbd.py (hunk 3 above)
Comment (bitkarrot) [reply lnbits/wallets/zbd.py:81]: the ZBD docs ask for "description" , not descriptionHash

see here: https://zbd.dev/api-reference/zbd-gamertag/create

**Comment by bitkarrot** [comment]
This PR should also fix the lightning address payments that were not working before.

File: lnbits/wallets/zbd.py (hunk 3 above)
Comment (dni) [reply lnbits/wallets/zbd.py:80]: i see a ```    "invoiceDescriptionHash": null ``` but on the response, the docs are weird, i think its fine to only send description to zbd

**Review by prusnak — APPROVED** [review]

File: lnbits/wallets/zbd.py (hunk 3 above)
Comment (bitkarrot) [reply lnbits/wallets/zbd.py:80]: OK

**Comment by bitkarrot** [comment]
@dni @motorina0  @prusnak Thank you

**Pull request merged**